signac/synced_collections/numpy_utils.py

```diff
@@ -32,7 +32,10 @@ def _convert_numpy(data):
     primitive types, while other numpy arrays are converted to lists. If ``data``
     is not a numpy data type, this function is a no-op.
     """
-    if NUMPY:
+    # Initially performing one isinstance check is faster since most of the
+    # time the inputs are not numpy arrays, preventing additional function
+    # calls inside the if statement.
+    if NUMPY and isinstance(data, (numpy.ndarray, numpy.number, numpy.bool_)):
```

Comment:
Can the second "elif" just be an "else", by elimination? Otherwise the "else" would be unreachable.

It's too far below this change for me to make a suggestion.

Reply:
Also should we support _all_ array scalars? We could use `numpy.generic` instead of `(numpy.number, numpy.bool_)`? You can have strings and dicts and other supported (JSON-encodable) types that are simply wrapped as array scalars. https://numpy.org/doc/stable/reference/arrays.scalars.html

Reply:
For your first question, yes, good call.

For your second question, I'm not opposed to doing that, but I'd rather punt on it for now. I intentionally added `np.bool_` here but not `flexible` or `object_` since I figured that we'd need to be much more thorough about testing those cases and deciding where errors should be thrown. We can definitely do this, I just don't want to think about it as part of #484.

Reply:
Sounds good.